pilot/pkg/bootstrap/certcontroller.go

```diff
@@ -110,7 +110,32 @@ func (s *Server) initDNSCerts(hostname, namespace string) error {
 
 	var certChain, keyPEM, caBundle []byte
 	var err error
-	if features.PilotCertProvider == constants.CertProviderKubernetes {
+	pilotCertProviderName := features.PilotCertProvider
+	if strings.HasPrefix(pilotCertProviderName, constants.CertProviderKubernetesSignerPrefix) && s.RA != nil {
```

Comment:
nit: don't think this should be related to whether the RA is enabled or not, we should be able to do this regardless

Reply:
yep. we can do this later.